### ISSUE TITLE
Normalizing contractions removes ending punctuation next to it

### ISSUE THREAD

**Issue by virustotalop** [spencermountain/compromise#589]
When normalizing this text

> "the guest-singer mr. smith who was supposed to show up at   at seven thirty didn't.

The output that is received is

> the guest singer Mr. Smith who was supposed to show up at at 7 30 did not

Expected output

> the guest singer Mr. Smith who was supposed to show up at at 7 30 did not.

Sample code

```javascript
var doc = nlp("the guest-singer mr. smith who was supposed to show up at   at seven thirty didn't.")
doc.normalize();
var topics = doc.topics();

var topicsArray = topics.out('array')
console.log(topics.out('array'));
for(var i = 0; i < topicsArray.length; i++)
{
	var topic = topicsArray[i];
	var match = topics.match(topic);
	match.toTitleCase();
}

console.log(doc.out());
```

This does not occur if contraction normalization is turned off.

**Comment by spencermountain** [comment]
ah! good bug!
thank you.

**Comment by virustotalop** [comment]
So I've looked into the bug further and I am able to reproduce a similar or the same bug with acronym and whitespace normalization.

Acronym normalization code
``` javascript
var doc = nlp("the guest-singer mr. smith who was supposed to show up at   at seven thirty    F.B.I.")
```

The output is 
```
the guest singer Mr. Smith who was supposed to show up at at 7 30 Fbi
```

Whitespace normalization code
```javascript
var doc = nlp("the guest-singer mr. smith who was supposed to show up at   at seven thirty   .")
```

Whitespace normalization on
```
the guest singer Mr. Smith who was supposed to show up at at 7 30
```

With whitespace normalization off
```
the guest-singer Mr. Smith who was supposed to show up at   at 7 30.
```

I haven't checked all normalization types yet but I do get the correct behavior from number normalization.

**Comment by spencermountain** [comment]
hey this situation has seems to have been considerably improved as of v12.
can you take a look?

```js
let doc = nlp(`the guest-singer mr. smith who was supposed to show up at at seven thirty didn't.`)
doc.normalize()
console.log(doc.text())
//the guest singer mr smith who was supposed to show up at at seven thirty didn't.

var doc = nlp('the guest-singer mr. smith who was supposed to show up at   at seven thirty    F.B.I.')
doc.normalize()
console.log(doc.text())
//the guest singer mr smith who was supposed to show up at at seven thirty FBI.
```
there's been some changes to normalization, documented [here](https://observablehq.com/@spencermountain/compromise-normalization) - notably number-parsing has been moved to a [seperate plugin](https://observablehq.com/@spencermountain/compromise-numbers)

by the way, i *really* love your AnonymizingText library! I've been wanting to do something similar for a long time.

**Comment by virustotalop** [comment]
@spencermountain 
Based on the output that you gave it seems that the bug is resolved. I don't have a proper environment to test the output but everything seems fixed to me, I'll make sure to make a report if I run into the bug in the future. Also thank you for the words of encouragement related to my AnoymizingText library, I wrote the project as my senior capstone project for the University that I attended. The code is a bit rough around the edges but I'm interested in re-writing it which is how I ran into the original bug that I reported. I also wanted to thank you for your project as it was invaluable when I wrote my capstone project.
